features-linux.md

```diff
@@ -0,0 +1,211 @@
+* **`systemd`** (bool, OPTIONAL) represents whether the runtime supports system-wide systemd cgroup manager.
+* **`systemdUser`** (bool, OPTIONAL) represents whether the runtime supports user-scoped systemd cgroup manager.
+* **`rdma`** (bool, OPTIONAL) represents whether the runtime supports RDMA cgroup controller.
+
+### Example
+
+```json
+"cgroup": {
+  "v1": true,
+  "v2": true,
+  "systemd": true,
+  "systemdUser": true,
+  "rdma": false
+}
+```
+
+## <a name="linuxFeaturesSeccomp" />Seccomp
+
+**`seccomp`** (object, OPTIONAL) represents the runtime's implementation status of seccomp.
+Irrelevant to the kernel version of the host operating system.
+
+* **`enabled`** (bool, OPTIONAL) represents whether the runtime supports seccomp.
+* **`actions`** (array of strings, OPTIONAL) The recognized names of the seccomp actions.
+  The runtime MUST recognize the elements in this array in the [`syscalls[].action` property of the `linux.seccomp` object in `config.json`](config-linux.md#seccomp).
+* **`operators`** (array of strings, OPTIONAL) The recognized names of the seccomp operators.
+  The runtime MUST recognize the elements in this array in the [`syscalls[].args[].op` property of the `linux.seccomp` object in `config.json`](config-linux.md#seccomp).
+* **`archs`** (array of strings, OPTIONAL) The recognized names of the seccomp architectures.
+  The runtime MUST recognize the elements in this array in the [`architectures` property of the `linux.seccomp` object in `config.json`](config-linux.md#seccomp).
+* **`knownFlags`** (array of strings, OPTIONAL) The recognized names of the seccomp flags.
+  The runtime MUST recognize the elements in this array in the [`flags` property of the `linux.seccomp` object in `config.json`](config-linux.md#seccomp).
+* **`supportedFlags`** (array of strings, OPTIONAL) The recognized and supported names of the seccomp flags.
+  This list may be a subset of `knownFlags` due to some flags not supported by the current kernel and/or libseccomp.
+  The runtime MUST recognize and support the elements in this array in the [`flags` property of the `linux.seccomp` object in `config.json`](config-linux.md#seccomp).
+
+### Example
+
+```json
+"seccomp": {
+  "enabled": true,
+  "actions": [
+    "SCMP_ACT_ALLOW",
+    "SCMP_ACT_ERRNO",
+    "SCMP_ACT_KILL",
+    "SCMP_ACT_LOG",
+    "SCMP_ACT_NOTIFY",
+    "SCMP_ACT_TRACE",
+    "SCMP_ACT_TRAP"
+  ],
+  "operators": [
+    "SCMP_CMP_EQ",
+    "SCMP_CMP_GE",
+    "SCMP_CMP_GT",
+    "SCMP_CMP_LE",
+    "SCMP_CMP_LT",
+    "SCMP_CMP_MASKED_EQ",
+    "SCMP_CMP_NE"
+  ],
+  "archs": [
+    "SCMP_ARCH_AARCH64",
+    "SCMP_ARCH_ARM",
+    "SCMP_ARCH_MIPS",
+    "SCMP_ARCH_MIPS64",
+    "SCMP_ARCH_MIPS64N32",
+    "SCMP_ARCH_MIPSEL",
+    "SCMP_ARCH_MIPSEL64",
+    "SCMP_ARCH_MIPSEL64N32",
+    "SCMP_ARCH_PPC",
+    "SCMP_ARCH_PPC64",
+    "SCMP_ARCH_PPC64LE",
+    "SCMP_ARCH_S390",
+    "SCMP_ARCH_S390X",
+    "SCMP_ARCH_X32",
+    "SCMP_ARCH_X86",
+    "SCMP_ARCH_X86_64"
+  ],
```

Comment:
how is the runtime supposed to query this list from libseccomp?  Is it expected to hardcode the list?

Reply:
> hardcode

Yes: https://github.com/opencontainers/runc/blob/v1.1.7/libcontainer/seccomp/config.go#L54-L71
But I guess it is not difficult to add a new function to libseccomp.

You can also keep the list nil.